chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```

Comment:
Please also consider this case:
![image](https://user-images.githubusercontent.com/26448672/230018911-a399a09a-04a0-4b03-aedb-87f31dc741f1.png)


I suppose that there should be output:
```
test/.test/
test/.test/1
test/.test/2
test/
test/1
test/2
```

Reply:
Do you think `1.txt.txt` should produce output `1`? Because sometimes it is not obvious if it is part of file name or not: `openwrt-22.03.0-ramips-mt7621-d-team_newifi-d2-squashfs-sysupgrade.bin`, `year.2023.txt`. What do you think?

Reply:
Discussed, changed to use only the first extension.